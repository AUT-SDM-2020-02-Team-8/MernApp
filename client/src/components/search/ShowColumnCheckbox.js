import React from "react"
import { Checkbox, FormControlLabel } from "@material-ui/core"

const ShowColumnCheckbox = (props) => {
  return (
    <FormControlLabel
      control={
        <Checkbox
          checked={props.chosenColumns.indexOf(props.name) >= 0}
          onChange={props.handleChange}
          name={props.name}
        />
      }
      label={props.label}
    />
  )
}

export default ShowColumnCheckbox