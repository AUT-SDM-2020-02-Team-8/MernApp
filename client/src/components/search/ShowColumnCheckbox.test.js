import React from 'react';
import { shallow } from 'enzyme';
import ShowColumnCheckbox from './ShowColumnCheckbox';

test('renders homepage', () => {
  const wrapper = shallow(<ShowColumnCheckbox
    name='title' label='Title' chosenColumns={['title']}
  />);
  expect(wrapper).toMatchSnapshot();
});
