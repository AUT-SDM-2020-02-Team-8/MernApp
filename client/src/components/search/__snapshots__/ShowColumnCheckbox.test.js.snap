// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage 1`] = `
<WithStyles(ForwardRef(FormControlLabel))
  control={
    <ForwardRef(WithStyles)
      checked={true}
      name="title"
    />
  }
  label="Title"
/>
`;
